internal/getproviders/registry_client.go

```diff
@@ -115,21 +106,23 @@ func (c *registryClient) ProviderVersions(addr addrs.Provider) ([]string, error)
 		return nil, nil
 	}
 
-	ret := make([]string, len(body.Versions))
-	for i, v := range body.Versions {
-		ret[i] = v.Version
+	ret := make(map[string][]string, len(body.Versions))
+	for _, v := range body.Versions {
+		ret[v.Version] = v.Protocols
 	}
 	return ret, nil
 }
 
-// PackageMeta returns metadata about a distribution package for a
-// provider.
+// PackageMeta returns metadata about a distribution package for a provider.
 //
-// The returned error will be ErrPlatformNotSupported if the registry responds
-// with 404 Not Found, under the assumption that the caller previously checked
-// that the provider and version are valid. It will return ErrUnauthorized if
-// the registry responds with 401 or 403 status codes, or ErrQueryFailed for
-// any other protocol or operational problem.
+// The returned error will be one of the following:
+//   - ErrPlatformNotSupported if the registry responds with 404 Not Found,
```

Comment:
I'm not 100% sure but I think godoc wants to see a blank link before these indented parts in order to treat them as preformatted text. I have a feeling that as currently written this would get rendered as a single paragraph with dashes inside it:

> The returned error will be one of the following: - ErrorPlatformNotSupport if the registry ...

(I have a feeling I'd originally wrote this out as a paragraph because godoc doesn't really support lists. I'm cool with using preformatted text as a substitute for a list though, because I agree that list formatting makes this much easier to scan.)